### ISSUE TITLE
Experimental details about your interesting paper 

### ISSUE THREAD

**Issue by mustansarfiaz** [mxbh/mapformer#1]
I found your paper titled "MapFormer: Boosting Change Detection by Using Pre-change Information " very interesting.  Could you please share experimental details and supplementary documents for this paper?

I wanted to reproduce the numbers, but I got very poor performance for the conditional BCD HRSCD dataset. Please find the log link and help me to reproduce the numbers.
https://drive.google.com/file/d/1PM9jO0wt0oc5JyjP_C-j-WmesmO_UmVY/view?usp=sharing

Here are the steps I performed.
I copied all the scripts from the tools folder to mapformer home directory.
Data preprocessing:
I noticed that for data, we need "-0M50-E080" added string only for tiles500 in the folder names of images, labels, and change. However, we don't need this for tiles2000
I run these lines 
           python convert_datasets/create_hrscd_tiles.py \
             --data_dir=./data/HRSCD/ \
             --out_dir=./data/HRSCD/preprocessed/tiles500_2 \
             --tile_size=500

Used for faster inference, internally splitted into 500px tiles within the model.
           python convert_datasets/create_hrscd_tiles.py \
             --data_dir=./data/HRSCD/ \
             --out_dir=./data/HRSCD/preprocessed/tiles2000_2 \
             --tile_size=2000
Training: I could not download the weights provided at this link https://drive.google.com/drive/folders/1b7bwrInTW4VLEm27YawHOAMSMikga2Ia.
However, I downloaded the weights at this link https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segformer/mit_b2_20220624-66e8bf70.pth and renamed them to  mit_b2.pth.
I modified the training script for the HRSCD dataset and ran this script.
           python train.py ./configs/conditional_bcd/hrscd/mapformer.yaml \
             --work-dir=./runs/conditional_bcd/hrscd/mapformer

Inference: very poor performance.

Please find the attached log file here https://drive.google.com/file/d/1PM9jO0wt0oc5JyjP_C-j-WmesmO_UmVY/view?usp=sharing
Please look into it and help me to reproduce the numbers.

Regards:
Mustansar Fiaz

**Comment by mxbh** [comment]
Hi,
thanks for your interest.
According to your logs, the keys of the state dict you used for initialization do not match the keys of the state dict of the model. Unfortunately, the GoogleDrive link to the pretrained checkpoints in the SegFormer repo is not valid anymore, but the OneDrive seems to works still. I updated the readme accordingly.

**Comment by mustansarfiaz** [comment]
Hi 

Thank you for your suggestion.  I followed as you recommended but the performance is very poor.

This time I used OneDrive but still, there was no improvement. Here is my log file: https://drive.google.com/file/d/1iT6dqyy1wBiqwgJtJ797rv6zsqYuojCA/view?usp=sharing

Regards:
Mustansar

**Comment by mxbh** [comment]
Hi,
looking at your provided logs, only the metrics regarding semantic segmentation are low and that is because the semantic segmentation head is not trained in the BCD-only setting. If you want to include the semantic segmentation head into training, use `./configs/conditional_scd/hrscd/mapformer_f2.yaml`  instead of `./configs/conditional_bcd/hrscd/mapformer.yaml`

**Comment by mustansarfiaz** [comment]
Hi 

Thank you for your feedback. So, as suggested, I run the code using` ./configs/conditional_scd/hrscd/mapformer_f2.yaml`
Here is my log file;
https://drive.google.com/file/d/1XbUSy4ku1Pb8NA6rFINmVC9H4TUU_nTu/view?usp=sharing

I got the numbers.  Now, my query is that the numbers reproduced in row 4 in the figure below belong to row 1 or row 2 or row 3. The numbers for row 1, row 2, and row 3 are copied from the paper tab. 4.


![image](https://github.com/mxbh/mapformer/assets/10016280/c1adf35f-7f35-40c1-828a-ed2ab39b4850)

Regards
Mustansar

**Comment by mxbh** [comment]
Hi ,
the config `./configs/conditional_scd/hrscd/mapformer_f2.yaml` corresponds to the first experiment in your screenshot, i.e., MapFormer( sem. seg. on f(2)).
